### PR TITLE
Add docs workflow for latest & testing

### PR DESCRIPTION
**Pull request by yunchu**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — APPROVED** [review]

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+name: Build Docs for the latest
+
+on:
+  workflow_dispatch: # run on request (no need for PR)
+  push:
+    branches:
+      - develop
```
Comment (harimkang) [inline .github/workflows/docs.yaml:7]: Is there any way to test docs before this v2 is replaced by develop?

Comment (yunchu) [reply .github/workflows/docs.yaml:7]: @harimkang only way to do that is merging this PR and update `docs.yml` in the develop to allow `workflow_dispatch` event and rename it from `docs.yml` to `docs.yaml`.

Comment (yunchu) [reply .github/workflows/docs.yaml:7]: @harimkang I've opened a PR for that. #3091

Comment (harimkang) [reply .github/workflows/docs.yaml:7]: Thanks!

**Review by eunwoosh — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**